### ISSUE TITLE
Add styles settings to some paragraphs types

### ISSUE THREAD

**Issue by herbdool** [backdrop-contrib/paragraphs_aplenty#27]
Inspired by `bootstrap_paragaphs`:

- `background` (none, basic colours (slight/light/strong) - black, bluegrey, blue, brown, cyan, green, grey, indigo, lime, orange, pink, purple, red, teal, white, yellow, transparent)
- `width` (none, tiny, narrow, medium, wide, full screen)
- `margin` (none, top and bottom (small/medium/large), top (small/medium/large), bottom (small/medium/large)
- `padding` (none, top and bottom (small/medium/large), top (small/medium/large), bottom (small/medium/large)

Alternative is to use https://github.com/backdrop-contrib/classy_paragraphs which gives people more flexibility. Could make a sub-module, or a button, to install some fields based on `classy_paragraphs` with preset colours, widths, margins padding.

**Comment by herbdool** [comment]
Reopening to add styles.

**Comment by herbdool** [comment]
@irinaz since you've been using paragraphs_aplenty, I'm wondering if you'd like to test this PR. It was mostly finished a few months ago but I held off merging it because I was hesitant of it conflicting with styles people may have without this installed. But since it's optional maybe I should just merge it.

It doesn't add all styles to each paragraph type, only where it makes sense. A column and column wrapper, for example, only has background field.

It will also conflict with the default Cards style, but someone could just delete the `background` and `plain` styles off of that list and only use the classy background field. I think it's best if I leave it for people to manage manually in case they've  made changes already.

**Comment by irinaz** [comment]
I will need some time to test it because we are using cards a lot both with Bootstrap5 theme and Decanter theme, and I need to make sure that new PR will not break our sites :)

**Comment by herbdool** [comment]
Sure, test away. Note that it is a separate submodule so no requirement to install.

**Comment by herbdool** [comment]
It won't override the default card background unless one is chosen from the list of new styles. I figured it should take precedence. Though things could be changed if it makes more sense a different way

**Comment by irinaz** [comment]
@herbdool , I want to confirm - I am testing latest code https://github.com/backdrop-contrib/paragraphs_aplenty that is not included in the most recent release, correct?   So I have two test cases, correct?

- install a module on a new site via UI, and it will have new code that I am testing
- download most recent code to a site that has module already installed and test that nothing is broken

**Comment by herbdool** [comment]
I wouldn't test this on an existing site, unless you've got a development environment for it, like in Lando.

It's in the PR, not in the main branch. So either patch the module with https://github.com/backdrop-contrib/paragraphs_aplenty/pull/35 or download the version of the module here: https://github.com/herbdool/paragraphs_aplenty/tree/issue-27b.

You'll need to enable the sub-module classy paragraphs aplenty. It requires the module https://github.com/backdrop-contrib/classy_paragraphs

**Comment by irinaz** [comment]
@herbdool , the reason I would want to use it on dev version of existing site is because I want to be sure that when this change makes into a release and I update the module on prod sites nothing will break or require a change. I will also coordinate with @ronan because we are using paragraphs with https://github.com/backdrop-contrib/stanford_decanter theme

**Comment by herbdool** [comment]
I've merged this. I've done a lot of testing and tweaked things before merging.

Note, that this feature is *optional*. It is a sub-module that can be enabled if someone wants to try it out.

**Comment by irinaz** [comment]
Sorry, I did not find time to test it properly.  Thanks for making this change, I will let you know if I see any issues.

**Comment by irinaz** [comment]
I finally did testing and noticed that text in cards is aligned to the bottom of the card - not sure if this was intentional or accidental 

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/0dbe24e9-5314-4521-b107-ee088059abc2" />


It comes from style 

.aplenty-card .field:nth-last-child(2) {
    flex: 1;
}

We added fix because we did not want text at the bottom, but I am reporting this just in case.  

@herbdool , please close this issue if that is intended style. 
Thanks a lot!

**Comment by herbdool** [comment]
@irinaz yes, that was intentional so that the buttons can line up on the bottom. I suggest you make a new issue where we can discuss. That feature was already in the module before this issue was merged so it's got nothing to do with the new Classy style fields.
